packages/esm-patient-chart-app/src/visit/past-visit-overview.component.tsx

```diff
@@ -0,0 +1,194 @@
+import { Visit } from '@openmrs/esm-api';
+import { createErrorHandler } from '@openmrs/esm-error-handling';
+import React, { useState, useEffect, useMemo, useReducer, useCallback } from 'react';
+import DataTable, {
+  TableContainer,
+  Table,
+  TableBody,
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableCell,
+  DataTableHeader,
+  DataTableRow,
+} from 'carbon-components-react/es/components//DataTable';
+import DataTableSkeleton from 'carbon-components-react/es/components/DataTableSkeleton';
+import OverflowMenu from 'carbon-components-react/es/components/OverflowMenu';
+import OverflowMenuItem from 'carbon-components-react/es/components/OverflowMenuItem';
+import { useTranslation } from 'react-i18next';
+import dayjs from 'dayjs';
+import { ErrorState } from '@openmrs/esm-patient-common-lib';
+import { attach, detach, getStartedVisit, getVisitsForPatient, VisitMode, VisitStatus } from '@openmrs/esm-framework';
+import styles from './past-visit-overview.component.scss';
+import Button from 'carbon-components-react/es/components/Button';
+
+// TO DO Abstract this logic for state machines to a hook
```

Comment:
I think that this is fine as-is, actually. If you want to, you can go ahead and trim down the interfaces, e.g. remove `Pending`, `Error` and `Resolved` in favor of:

```ts
interface Action {
  type: ActionTypes;
}
```